exp/api/v1beta1/machinepool_webhook.go

```diff
@@ -63,6 +65,12 @@ func (m *MachinePool) Default() {
 	if m.Spec.Template.Spec.InfrastructureRef.Namespace == "" {
 		m.Spec.Template.Spec.InfrastructureRef.Namespace = m.Namespace
 	}
+
+	// tolerate version strings without a "v" prefix: prepend it if it's not there.
+	if m.Spec.Template.Spec.Version != nil && !strings.HasPrefix(*m.Spec.Template.Spec.Version, "v") {
+		normalizedVersion := "v" + *m.Spec.Template.Spec.Version
```

Comment:
Just to double-check: is this triggering rollouts on existing infrastructure or not?

Reply:
Only if the user created a MP without the v prefix (which as described in https://github.com/kubernetes-sigs/cluster-api/issues/4651 isn't functional). Once the template version has the v prefix it won't modify it / cause a rollout.